### PR TITLE
feat: add support for min_score on ScriptScoreQuery

### LINKED ISSUE #622: [BUG] No MinScore property exposed on ScriptScore query
### What is the bug?
The docs specify an optional MinScore property https://opensearch.org/docs/latest/query-dsl/specialized/script-score/

Unless I am missing something, there is no method to specify this in opensearch.client

### How can one reproduce the bug?
``` var kNNQueryDescriptor = new ScriptScoreQueryDescriptor<Product>();

 if (searchCriteria.Embedding == null || searchCriteria.Embedding.Length == 0)
 {
     return kNNQueryDescriptor;
 }

 kNNQueryDescriptor
     .Query(q => q.Bool(x => x.Must(y => y.Term(f => f.SoldOut, false))))
     .Script(s => s.Source("knn_score")
                   .Lang("knn")
                   
                   .Params( new Dictionary<string, object> {
                       { "field", "searchableTextEmbedding" }, 
                       { "query_value", searchCriteria.Embedding }, 
                       { "space_type", "cosinesimil" } 
                   }));

 return kNNQueryDescriptor;
```
 
 Where in above can I set the MinScore property?

### What is the expected behavior?
_A clear and concise description of what you expected to happen._

### What is your host/environment?
Windows 11

### Do you have any screenshots?
image
image
image

### Do you have any additional context?
_Add any other context about the problem._

### PR DESCRIPTION
**Pull request by SamuelCox**
### Description
Add support for `min_score` property in `ScriptScoreQuery`. This also is a proper version of https://github.com/opensearch-project/opensearch-net/pull/623 which I closed after realising trying to add something to an SDK in the github browser editor was probably a bad idea.

Also, when running `build.bat` locally I get a few errors, I don't want to take the action minutes up with another duff run, so could someone check if this is just something not correctly set up on my machine, or an actual error I've caused?
image
image
I could be missing it, but I couldn't find a guide on how to run these tests so I'm assuming I'm missing some setup

### Issues Resolved
Closes #622 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by SamuelCox** [comment]
I can mark this as ready for review if the maintainers are happy, I just thought I should wait to figure out why those tests are failing first.

**Comment by SamuelCox** [comment]
Have marked as ready for review so it gets in front of maintainers, but like I say I'm not sure what's causing those test failures on my local machine

**Comment by SamuelCox** [comment]
image
Seems like those NullRefs are gone after trying a couple more times, so I guess they were just flaky.

**Comment by dblock** [comment]
@tfarrep where does this fall in the code generation?

@SamuelCox check whether this field exists in https://github.com/opensearch-project/opensearch-api-specification, please and contribute there if it's not, we're trying to get most code generated from spec

@SamuelCox on the errors, I think we need a developer guide like https://github.com/opensearch-project/opensearch-py/blob/main/DEVELOPER_GUIDE.md (want to start one?) and we can debug what's going on with your own setup

File: CHANGELOG.md
Comment (dblock) [inline CHANGELOG.md:10]: Nit: remove empty line.

**Comment by SamuelCox** [comment]
> @tfarrep where does this fall in the code generation?
> 
> 
> 
> @SamuelCox check whether this field exists in https://github.com/opensearch-project/opensearch-api-specification, please and contribute there if it's not, we're trying to get most code generated from spec
> 
> 
> 
> @SamuelCox on the errors, I think we need a developer guide like https://github.com/opensearch-project/opensearch-py/blob/main/DEVELOPER_GUIDE.md (want to start one?) and we can debug what's going on with your own setup
> 
> 
> 
> 
image
It appears to exist in the spec already if I understand correctly. @Xtansia asked me to add it here in the issue I logged but perhaps I misunderstood.

I agree that a dev guide would be good, probably would be best to add as a separate PR though?

Happy to add a follow up when I get time

**Comment by Xtansia** [comment]
@dblock This is a field in the request body and the generator does not yet generate the request/response body structures. 

@SamuelCox Thanks for PR'ing this so quickly, changes look good to me, just waiting on the CI to pass. Unfortunately there is a couple flaky tests floating around, I've made an issue regarding your specific null ref one (#626). The TestEphemeralCluster 403 errors are ones I haven't experienced myself (nor seen in CI), but were also reported in #282. So if you happen to gain any further insight please add it to the relevant issues. I also second that if you wanted to contribute towards a developer's guide that'd be amazing!

**Review by Xtansia — APPROVED** [review]

**Pull request merged**
